### PR TITLE
regmgr: refactor and fix int value.

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- refactor int value for get saved it in 4 byte in file
- add type of value and using it for no set value if is 0 on str/bin.

credit to @Princess-of-Sleeping for founded this type of value.
and to @bookmist for reverse byte for can re get int.

### CODE REVIEW

File: vita3k/regmgr/src/regmgr.cpp [hunk 1]
```diff
@@ -58,15 +58,41 @@ static std::string decryptRegistryFile(const fs::path reg_path) {
...
+    // Convert the hex value to a hex string (bytes)
+    std::string res;
+    for (size_t i = 0; i < hex.size(); i += 2) {
+        const int byte_value = std::stoi(hex.substr(i, 2), nullptr, 16);
+        res += static_cast<char>(byte_value);
+    }
+
+    return std::vector<char>(res.begin(), res.end());
+}
+
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:95]: wtf is that?

File: vita3k/regmgr/src/regmgr.cpp [hunk 2]
```diff
@@ -347,15 +392,17 @@ int32_t get_int_value(RegMgrState &regmgr, const std::string &category, const st
...
+    // Reverse the bytes
+    std::vector<char> reversed(4);
+    for (int i = 0; i < 4; i++)
+        reversed[i] = sys[3 - i];
+
+    // Convert to int with the reversed bytes
+    int value;
+    std::copy(reversed.begin(), reversed.end(), reinterpret_cast<char *>(&value));
+
+    return value;
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:405]:
```suggestion
    return byte_swap(std::bit_cast<uint32_t>(sys.data()));
```
Also include the util headers for this to work.

File: vita3k/regmgr/src/regmgr.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:95]: where is problem ???

File: vita3k/regmgr/src/regmgr.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:405]: is see

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: that is done.

File: vita3k/regmgr/src/regmgr.cpp (hunk 1 above)
Comment (Macdu) [reply vita3k/regmgr/src/regmgr.cpp:96]:
```suggestion
static std::vector<char> convert_int_to_hex_string(const int32_t &value) {
    std::vector<char> res(4);
    *reinterpret_cast<int32_t*>(res.data()) = byte_swap(value);
    return res;
}

```

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: Is see, i no know project have this func for that
thx

**Comment by Zangetsu38** [comment]
all is done

File: vita3k/regmgr/src/regmgr.cpp [hunk 3]
```diff
@@ -58,15 +59,32 @@ static std::string decryptRegistryFile(const fs::path reg_path) {
...
+    RegType type;
     uint32_t size;
-    std::string value;
+    std::vector<char> init_value;
 };
 
 static std::vector<std::string> reg_category_template;
 static std::map<std::string, std::vector<RegValue>> reg_template;
 
+static std::vector<char> convert_int_to_hex_string(const int32_t &value) {
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:78]:
```suggestion
static std::vector<char> convert_int_to_hex_string(int32_t value) {
```
Don't pass an int by reference

File: vita3k/regmgr/src/regmgr.cpp [hunk 4]
```diff
@@ -58,15 +59,32 @@ static std::string decryptRegistryFile(const fs::path reg_path) {
...
+    // Aloocate size for 4 bytes
+    std::vector<char> res(4);
+
+    // Convert the int to a hex string (4 bytes)
```
Comment (Macdu) [inline vita3k/regmgr/src/regmgr.cpp:82]: That's definitely not a hex string

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:82]: is hex string
image
Is hex value writed in char

File: vita3k/regmgr/src/regmgr.cpp (hunk 3 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: ah yes

File: vita3k/regmgr/src/regmgr.cpp (hunk 4 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: if you log this result to you get this .5p., is hexa value converted to char

File: vita3k/regmgr/src/regmgr.cpp [hunk 5]
```diff
@@ -346,26 +372,19 @@ int32_t get_int_value(RegMgrState &regmgr, const std::string &category, const st
...
-    if (!std::all_of(value_str.begin(), value_str.end(), [](unsigned char c) { return std::isdigit(c) || (c == '\0'); })) {
-        LOG_ERROR("Invalid value for {}{}: {}, attempt using default value!", cat, name, value_str);
-        value_str = set_default_value(regmgr, cat, name);
-        if (value_str.empty())
-            return 0;
-    }
 
-    return string_utils::stoi_def(value_str);
+    // Convert hex string to int
+    return byte_swap(*reinterpret_cast<const uint32_t *>(regmgr.system_dreg[cat][name].data()));
```
Comment (bookmist) [inline vita3k/regmgr/src/regmgr.cpp:377]: Byte swap do not do anything with hex. It switch endianness. Delete your comment, it's misleading. (same further)

File: vita3k/regmgr/src/regmgr.cpp (hunk 4 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: Code litteraly convert int number to

File: vita3k/regmgr/src/regmgr.cpp (hunk 5 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp:377]: OK if you said

File: vita3k/regmgr/src/regmgr.cpp (hunk 4 above)
Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: i have removed comment but like i said, i have just follow what princess have learn me for convert binary hexa char to hexa value
is same
for screen category verison is 
decimale = 3502084
hexa = 00 35 70 04 you can see it on screen
char writed .p5.

Comment (Macdu) [reply vita3k/regmgr/src/regmgr.cpp]: That has absolutely nothing to do with hexadecimal and that's not even a string. It's just that the data is stored as big endian

Comment (Zangetsu38) [reply vita3k/regmgr/src/regmgr.cpp]: ok, if you said👌

**Review by bookmist — APPROVED** [review]

**Pull request merged**